internal/controller/clusterpolicy_controller.go

```diff
@@ -0,0 +1,186 @@
+)
+
+// PolicyReconciler reconciles a AutomatedException object
+type ClusterPolicyReconciler struct {
+	client.Client
+	EdgeDBClient     *edgedb.Client
+	Scheme           *runtime.Scheme
+	MaxJitterPercent int
+}
+
+var (
+	//GiantSwarm team label
+	GSTeamLabel = "application.giantswarm.io/team"
+	//Policy API Exemption label
+	//TODO: Move to Policy API
+	PolicyAPIExemptionLabel = "policy.giantswarm.io/giantswarm-exempt"
+)
+
+func (r *ClusterPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	var clusterPolicy kyvernoV1.ClusterPolicy
+
+	if err := r.Get(ctx, req.NamespacedName, &clusterPolicy); err != nil {
+		if !errors.IsNotFound(err) {
+			// Error fetching the report
+			log.Log.Error(err, "unable to fetch Kyverno ClusterPolicy")
+		} else {
+			// Policy not found, make sure we don't have it in edgedb either
+			err := edgedbutils.DeleteKyvernoClusterPolicy(ctx, r.EdgeDBClient, req.Name)
+			if err != nil {
+				log.Log.Error(err, "Error deleting ClusterPolicy from database")
+			}
+		}
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if clusterPolicy.Spec.ValidationFailureAction.Enforce() {
+		if clusterPolicy.HasValidate() {
+			log.Log.Info("Kyverno ClusterPolicy is in enforce mode and has validating rules, adding it to edgedb")
+
+			policyRuleNames := extractRuleNames(clusterPolicy)
+			policyTargetKinds := extractTargetKinds(clusterPolicy)
+			gsExempt := shouldExcludeGiantSwarmResources(clusterPolicy)
+
+			if len(policyRuleNames) == 0 || len(policyTargetKinds) == 0 {
+				log.Log.Error(errors.New("Error extracting rule names or target kinds"), fmt.Sprintf("Error extracting rule names or target kinds from Kyverno ClusterPolicy %s", clusterPolicy.Name))
+				return ctrl.Result{}, nil
+			}
+
+			_, err := edgedbutils.InsertKyvernoClusterPolicy(ctx, r.EdgeDBClient, clusterPolicy.Name, policyRuleNames, policyTargetKinds, gsExempt)
+			if err != nil {
+				log.Log.Error(err, "Error inserting Kyverno ClusterPolicy in database")
+			}
+
+		}
+	}
+
+	return utils.JitterRequeue(utils.DefaultRequeueDuration, r.MaxJitterPercent, log.Log), nil
+}
+
+func shouldExcludeGiantSwarmResources(clusterPolicy kyvernoV1.ClusterPolicy) bool {
+	gsExempt := true
+
+	// Check if team label exist
+	if _, ok := clusterPolicy.Labels[GSTeamLabel]; ok {
+		// If team label exists, this policy comes from Giant Swarm, so our workloads are not exempt; they should satisfy the policy, ship an exception, or be excluded within the policy itself.
+		gsExempt = false
+	}
+
+	// Check if the policy has a label enabling or disabling GS exemption.
+	if gsExemptLabelValue, ok := clusterPolicy.Labels[PolicyAPIExemptionLabel]; ok {
+		var err error
+
+		gsExempt, err = strconv.ParseBool(gsExemptLabelValue)
+		if err != nil {
+			// The label value is garbage. Complain and error out, or default the behavior
+			// Label is probably not set, set default to false
```

Comment:
Probably doesn't make a difference here, but we know the label is set because it passed `ok` (right?)

Reply:
Ah I totally forgot about that, it's true